src/Azure.Functions.Cli/Common/AuthManager.cs

```diff
@@ -0,0 +1,312 @@
+            MiddlewareAuthSettings.SetAuthSetting("WEBSITE_AUTH_TOKEN_STORE", "True");
+            MiddlewareAuthSettings.SetAuthSetting("WEBSITE_AUTH_UNAUTHENTICATED_ACTION", "AllowAnonymous");
+
+            // Middleware requires signing and encryption keys for local testing
+            // These will be different than the encryption and signing keys used by the application in production
+            string encryptionKey = ComputeSha256Hash(clientSecret);
+            string signingKey = ComputeSha256Hash(clientId);
+            MiddlewareAuthSettings.SetAuthSetting("WEBSITE_AUTH_ENCRYPTION_KEY", encryptionKey);
+            MiddlewareAuthSettings.SetAuthSetting("WEBSITE_AUTH_SIGNING_KEY", signingKey);
+            MiddlewareAuthSettings.SetAuthSetting("WEBSITE_AUTH_ALLOWED_AUDIENCES", serializedReplyUrls);
+            MiddlewareAuthSettings.Commit();
+        }
+
+        public Dictionary<string, string> CreateAuthSettingsToPublish(string homepage, string clientId, string clientSecret, string tenant, List<string> replyUrls)
+        {
+            // The 'allowedAudiences' setting of /config/authsettings is of the form ["{replyURL1}", "{replyURL2}"]
+            string serializedArray = JsonConvert.SerializeObject(replyUrls, Formatting.Indented);
+
+            var authSettingsToPublish = new Dictionary<string, string>();
+            authSettingsToPublish.Add("allowedAudiences", serializedArray);
+            authSettingsToPublish.Add("isAadAutoProvisioned", "True");
+            authSettingsToPublish.Add("clientId", clientId);
+            authSettingsToPublish.Add("clientSecret", clientSecret);
+            authSettingsToPublish.Add("defaultProvider", "0"); // 0 corresponds to AzureActiveDirectory
+            authSettingsToPublish.Add("enabled", "True");
+            authSettingsToPublish.Add("issuer", "https://sts.windows.net/" + tenant + "/");
+            authSettingsToPublish.Add("runtimeVersion", "1.0.0");
+            authSettingsToPublish.Add("tokenStoreEnabled", "True");
+            authSettingsToPublish.Add("unauthenticatedClientAction", "1"); // Corresponds to AllowAnonymous
+
+            return authSettingsToPublish;
+        }
+    
+        private static async Task<bool> PublishAuthSettingAsync(Site functionApp, string accessToken, Dictionary<string, string> authSettings)
+        {
+            functionApp.AzureAuthSettings = authSettings;
+            var result = await AzureHelper.UpdateFunctionAppAuthSettings(functionApp, accessToken);
+            if (!result.IsSuccessful)
+            {
+                ColoredConsole
+                    .Error
+                    .WriteLine(ErrorColor("Error updating app settings:"))
+                    .WriteLine(ErrorColor(result.ErrorResult));
+                return false;
+            }
+            return true;
+        }
+
+        static string ComputeSha256Hash(string rawData)
```

Comment:
You can add this method to [`StringExtensions.cs`](https://github.com/Azure/azure-functions-core-tools/blob/master/src/Azure.Functions.Cli/Extensions/StringExtensions.cs)